### PR TITLE
Validate domains

### PR DESCRIPTION
**Pull request by gastonfig**
This PR adds the `PendingDomains` components to allow admins to approve, deny, and add new domains. This component replaces the removed `DomainValidation.js` file.

Please note the creation of a new method `setPendingDomain` in `firebaseBackend.js`. I couldn't find a better way to accomplish this but please let me know if there's a better approach.

### CODE REVIEW

File: src/components/PendingDomains.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  handleApproveDomain(domain) {
+    this.props.backend.setDomainIsValid(domain, true)
+  }
+
+  handleDenyDomain(domain) {
+    this.props.backend.setDomainIsValid(domain, false)
+  }
+
+  addDomain(event) {
+    event.preventDefault()
```
Comment (newhouseb) [inline src/components/PendingDomains.js]: you should just be able to call setDomainIsValid, no need to go through the intermediate step

**Comment by newhouseb** [comment]
Can we make the columns disappear once they've been successfully changed? I thought things weren't working because nothing seemed to change when i clicked the buttons.

File: src/components/PendingDomains.js (hunk 1 above)
Comment (gastonfig) [reply src/components/PendingDomains.js]: Great point

File: src/lib/firebaseBackend.js [hunk 2]
```diff
@@ -452,17 +452,28 @@ class FirebaseBackend extends BackendInterface {
 
   // VALIDATED DOMAINS
 
-  async getDomains(pendingOnly) {
+  async getDomains(pendingOnly, callback) {
```
Comment (gastonfig) [inline src/lib/firebaseBackend.js]: @newhouseb I tried to return the data (`newDomains`) with this function but it wouldn't trigger renders in the component so ended up passing a callback which passes the data.

**Pull request merged**